### PR TITLE
[APT-2001] Support stage environment cart

### PR DESCRIPTION
**Pull request by oredavids**
[APT-2001]

## Changes
- Added default & prod site config for `developer_portal_url`
- Replaced `site.data.products` usage with `site.products` to enable correct configuration for development and production environments

### Context
Jekyll supports [variable declarations](https://jekyllrb.com/docs/variables/) and our site-wide configurations are set in default `./_config.yml`. We already have deployment scripts `./push-to-s3-{ENV}.sh` that combine the right `./_config.{ENV}.yml` with the default config file.

#### Pre-requisite
- [x] Review, merge this [Dev Portal PR](https://github.com/gruntwork-io/aperture/pull/1009)

#### Steps
- [x] Go the Preview environment [checkout page](https://deploy-preview-746--keen-clarke-470db9.netlify.app/checkout/)
- [x] Select one or more available add-ons and click the checkout button
- [x] Observe that you have been redirected to the checkout page of the stage developer portal app
- [x] Observe that all your product selections are reflected on the checkout page

#### Pre-requisite
- [ ] Ensure the changes in this [Dev Portal PR](https://github.com/gruntwork-io/aperture/pull/1009) have been released to the production developer portal app
- [ ] Release a new version of this website after this PR has been merged

#### Steps
- [ ] Go the Production environment [checkout page](https://gruntwork.io/checkout)
- [ ] Select one or more available add-ons and click the checkout button
- [ ] Observe that you have been redirected to the checkout page of the production developer portal app
- [ ] Observe that all your product selections are reflected on the checkout page

[APT-2001]: https://gruntwork.atlassian.net/browse/APT-2001?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: assets/js/checkout.js [hunk 1]
```diff
@@ -75,8 +75,11 @@ $(function () {
     // Auto toggles addons based on the URI
     const rawCart = atob($.query.get("cart"));
 
+    console.log("rawCart", rawCart);
+
```
Comment (ebeneliason) [inline assets/js/checkout.js:79]:
```suggestion
```

File: assets/js/checkout.js [hunk 2]
```diff
@@ -75,8 +75,11 @@ $(function () {
...
     if (rawCart) {
       const products = JSON.parse(rawCart);
+      console.log("products", products);
```
Comment (ebeneliason) [inline assets/js/checkout.js:82]:
```suggestion
```

**Review by ebeneliason — COMMENTED** [review]
This looks nice and small and relatively straightforward, but I have a couple of questions:
1. Is there a reason that the stage environment file isn't `_config.stage.yml` to mirror prod?
2. What machinery determines which config to load? Did I miss it?

**Comment by oredavids** [comment]
@ebeneliason 

> Is there a reason that the stage environment file isn't _config.stage.yml to mirror prod?

There is a `_config.stage.yml` already in the codebase. There is no change to this file in this PR because the default config in `_config.yml` is sufficient for local dev, previews, and stage environments. The prod file overwrites the default with its own prod values.

> What machinery determines which config to load? Did I miss it?

The website deployments to Stage and Prod are handled by CircleCI. See the scripts being called [here](https://github.com/gruntwork-io/gruntwork-io.github.io/blob/40ed044d76c6dad01b391e60ef7c3ac8c3c83083/.circleci/config.yml#L62) and [here](https://github.com/gruntwork-io/gruntwork-io.github.io/blob/40ed044d76c6dad01b391e60ef7c3ac8c3c83083/.circleci/config.yml#L77) in our CircleCI config file. If you scroll to the end of the config file; you'll see that the `stage` job is invoked on every commit to the master branch while the `prod` job is invoked when there is a new tag.

**Review by ebeneliason — APPROVED** [review]
Looks good!

**Pull request merged**